views/js/pciCreator/dev/audioRecordingInteraction/runtime/js/recorder.js

```diff
@@ -134,8 +133,26 @@ define([
          */
         function getInputLevel() {
             var sum;
+            var frequencyLength;
+            var getRndInteger;
 
             analyser.getByteFrequencyData(frequencyArray);
+            frequencyLength = frequencyArray.length;
+
+            if (frequencyArray.length && /(iPhone|iPad)/i.test(navigator.userAgent)) {
+                getRndInteger = function(min, max) {
+                    return Math.floor(Math.random() * (max - min) ) + min;
+                };
+
+                frequencyArray[0] = getRndInteger(0, frequencyLength * frequencyLength);
+
+                _.forEach(frequencyArray, function(level, index) {
+                    frequencyArray[index] = frequencyArray[0];
+                    if (frequencyLength / 2 < index) {
```

Comment:
This condition looks weird.
And what will happen to the remaining part of the array?

Reply:
`frequencyArray` is filled with zero values before the loop.
So the aim is to fill half of it with values and leave the rest as it is (zeroes).
This manipulation let us to imitate the green frequency line to fill half of the scale.